src/app/api/places/chat/route.js

```diff
@@ -0,0 +1,121 @@
+import { generateText } from "ai";
+import { createOllama } from "ollama-ai-provider";
+
+const generateGoogleFindUrlUsingLLM = async (userLocationString, messages) => {
+  const llama = createOllama();
+  return generateText({
+    model: llama("openhermes2.5-mistral"),
+    system: `Your only task is to construct a valid Google Find Place API URL. 
+    You are allowed to use information provided in system prompt and the conversation.
+
+    Additional information:
+    ${userLocationString}
+    
+    Additional guidelines you should follow:
+      - The URL should be wrapped between three backticks
+      - Use radius of 2000 if it can't be determined from the conversation
+      - locationbias: A string specifying radius in meters, plus latitude/longitude in decimal degrees. Use the following format: circle:radius@latitude,longitude
+      - The Google API key is ${process.env.GOOGLE_API_KEY}
+      - inputtype is textquery
+      - The fields you should include are: formatted_address, name, rating, opening_hours, geometry
+      - The input should be parsed from the conversation.
+
+    Example of a correctly generated URL:
+      - https://maps.googleapis.com/maps/api/place/findplacefromtext/json?fields=formatted_address%2Cname%2Crating%2Copening_hours%2Cgeometry&input=mongolian&inputtype=textquery&locationbias=circle:2000@47.6918452,-122.2226413&key=${process.env.GOOGLE_API_KEY}
```

Comment:
We can insert key manually, there is no point of using unreliable AI to pass the key back and forth!